src/app/qgsrelationaddpolymorphicdialog.cpp

```diff
@@ -94,18 +90,19 @@ void QgsRelationAddPolymorphicDialog::setPolymorphicRelation( const QgsPolymorph
   mReferencedLayerExpressionWidget->setExpression( polyRel.referencedLayerExpression() );
   mRelationStrengthComboBox->setCurrentIndex( mRelationStrengthComboBox->findData( polyRel.strength() ) );
 
+  const QStringList layerIds = polyRel.referencedLayerIds();
+  for ( const QString &layerId : layerIds )
+    mReferencedLayersComboBox->setItemCheckState( mReferencedLayersComboBox->findData( layerId ), Qt::Checked );
+  referencedLayersChanged();
+
   int index = 0;
   const QList<QgsRelation::FieldPair> fieldPairs = polyRel.fieldPairs();
   for ( const QgsRelation::FieldPair &fieldPair : fieldPairs )
   {
-    static_cast<QLineEdit *>( mFieldsMappingTable->cellWidget( index, 0 ) )->setText( fieldPair.referencedField() );
+    static_cast<QComboBox *>( mFieldsMappingTable->cellWidget( index, 0 ) )->setCurrentText( fieldPair.referencedField() );
```

Comment:
For such down casts, the intent is better captured with:
```suggestion
    qgis::down_cast<QComboBox *>( mFieldsMappingTable->cellWidget( index, 0 ) )->setCurrentText( fieldPair.referencedField() );
```
Similarly for line below